### PR TITLE
Lists of fitting results

### LINKED ISSUE #276: Simple list views of fitting results
The simplest way to navigate to and view results of individual fitting experiments is just to list everything linked to a given model/protocol/dataset/fitting spec version.

There's commented out code in `entity_version.html` for a button that could link to such a view. Though the button text should probably read 'Fitting results'.

The view itself will be very similar to `entities/compare_experiments.html`, except that initially it won't have the comparison checkboxes, just the individual links. (Until the comparison views from #134 are done.)

- [x] List all fitting results for a given model version
    - Group the list by dataset
- [x] List all fitting results for a given dataset
    - Group the list by model
    - Have a button to show all versions or only the latest
- [x] List all fitting results for a given protocol
    - Group by dataset, subgroup by model
    - Have a button to show all model versions or only the latest
- [x] List all fitting results for a given fitting spec
    - Group by dataset, subgroup by model
    - Have a button to show all model versions or only the latest

Child of #135.

### PR DESCRIPTION
**Pull request by helenst**
Fixes #276

### CODE REVIEW

File: weblab/templates/datasets/compare_fittings.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
+{% extends "base.html" %}
+{% load entities %}
+{% load fittings %}
+{% load staticfiles %}
+
+{% block title %}Dataset : compare fitting experiments - {% endblock title %}
+
+{% block content %}
```
Comment (jonc125) [inline weblab/templates/datasets/compare_fittings.html:8]: There's a lot of overlap here with the dataset_detail template. Could we make a new dataset_base or similar that both inherit from maybe?

File: weblab/entities/tests/test_views.py [hunk 2]
```diff
@@ -652,6 +652,290 @@ def test_returns_404_if_commit_not_found(self, client, logged_in_user):
...
+            '/entities/models/%d/versions/%s/fittings' % (fit.model.pk, fit.model_version.sha)
+        )
+
+        assert response.status_code == 200
+        assert response.context['comparisons'] == [(fit.dataset, [fit])]
+
+    def test_groups_by_dataset_for_model(self, client, public_model):
+        model_version = public_model.repocache.latest_version
+        dataset1, dataset2 = recipes.dataset.make(_quantity=2, visibility='public')
+
```
Comment (jonc125) [inline weblab/entities/tests/test_views.py:676]: Similarly to above, can we add some private versions and check the corresponding results aren't included?

**Review by jonc125 — CHANGES_REQUESTED** [review]
Looking good. I tweaked some aspects of the display; figured it was easier just to make the changes than put them as review comments. There are just a few minor things to finish off I think.

**Review by jonc125 — APPROVED** [review]
Just a very minor suggestion!

I was confused at first that the visibility form still wasn't showing for me (hence the extra test added) but killing & restarting my dev server resolved it 🤷‍♂️

File: weblab/datasets/tests/test_views.py [hunk 3]
```diff
@@ -730,3 +730,41 @@ def test_multiple_model_versions_for_dataset(self, client, helpers, public_datas
...
             (m2, [fit3_m2v]),
         ]
+
+    def test_ensure_private_results_are_not_shown(self, client, public_dataset):
+        recipes.fittingresult_version.make(
+            fittingresult__dataset=public_dataset,
+            fittingresult__model_version__visibility='private',
+            fittingresult__protocol_version__visibility='public',
+            fittingresult__fittingspec_version__visibility='public',
+        ).fittingresult
```
[the remark this answers is not in the document]
Comment (helenst) [reply weblab/datasets/tests/test_views.py]: Indeed - have fixed this, and similarly in entity view tests.

**Pull request merged**